### ISSUE TITLE
Color doesn't work

### ISSUE THREAD

**Issue by stasbar** [PaoloRotolo/AppIntro#94]
Hello.
I can't change slide background color i have tried AppIntro and AppIntro2
My code:
````
public class MyIntro extends AppIntro {
    @Override
    public void init(Bundle savedInstanceState) {
        addSlide(AppIntroFragment.newInstance(getString(R.string.intro_title_main)
                , getString(R.string.intro_description_main), R.drawable.intro_image_main
                , R.color.colorAccent));
        showDoneButton(true);
        setFlowAnimation();
    }
````
where R.color.colorAccent is light blue
````
<color name="colorAccent">#2480b3</color>
````
manifest 
````
<activity
            android:name=".activity.MyIntro"
            android:label="@string/app_name"
            android:theme="@style/Intro">
````
styles
````
<style name="Intro" parent="Theme.AppCompat.NoActionBar">
</style>
````
and the result: 
![device-2015-10-12-002847](https://cloud.githubusercontent.com/assets/9407590/10419430/4b6a82d4-7078-11e5-9b82-ba2a3c534d16.png)

**Comment by FallenRiteMonk** [comment]
I have the same problem. The only way I can change the color is by passing a ColorInt from the Color class.
exp:
```
addSlide(AppIntroFragment.newInstance(getString(R.string.guide), getString(R.string.guide_title), R.mipmap.icon, Color.RED));
```

**Comment by ja1984** [comment]
try this:

```java
addSlide(AppIntroFragment.newInstance(getString(R.string.intro_title_main)
                , getString(R.string.intro_description_main), R.drawable.intro_image_main
                , getResources().getColor(R.color.colorAccent));
```

You will get a depricated warning but it works!

**Comment by stasbar** [comment]
My workaround:
Clone AppIntroFragment and change onCreate to
````
@Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View v = inflater.inflate(com.github.paolorotolo.appintro.R.layout.fragment_intro, container, false);
        TextView t = (TextView) v.findViewById(com.github.paolorotolo.appintro.R.id.title);
        TextView d = (TextView) v.findViewById(com.github.paolorotolo.appintro.R.id.description);
        ImageView i = (ImageView) v.findViewById(com.github.paolorotolo.appintro.R.id.image);
        LinearLayout m = (LinearLayout) v.findViewById(com.github.paolorotolo.appintro.R.id.main);
        t.setText(title);
        d.setText(description);
        i.setImageResource(drawable);
        m.setBackgroundResource(colour);
        return v;
    }
````
and in init:
````
addSlide(SampleSlide.newInstance(getString(R.string.intro_title_main)
                , getString(R.string.intro_description_main), R.drawable.intro_image_main
                , R.color.colorAccent));
````
Works perfectly ;)

**Comment by Andrew-Quebe** [comment]
@stasbar please consider making a PR with your workaround.

**Comment by stasbar** [comment]
Ok but, What is PR ?

**Comment by Andrew-Quebe** [comment]
@stasbar it's a pull request. You basically fork all the code, make your changes to the lib, and then submit a pull request. I'll review it and merge it with the project.

**Comment by HeinrichReimer** [comment]
As this is a very serious problem for developers in terms of debugging it would be great if @PaoloRotolo  and @Andrew-Quebe used androids annotations:

- `@ColorInt` for colors (`0xFF00FF00`)
- `@ColorRes` for color resource ids (`R.color.your_color`)
- `@StringRes` for string resource ids (`R.string.your_string`)
- `@DrawableRes` for drawable resource ids (`R.drawable.your_image`)

Only problem I see is that there seem to be issues with android annotations as described in #105 which can easily be solven by solving #119 ;)

**Comment by porokoro** [comment]
**TL;DR;** @HeinrichReimer you can use the annotations you mentioned without problems.

Hello @HeinrichReimer, I think there is a difference between [Android Annotations](https://github.com/excilys/androidannotations) and the support annotations (often also called android annotations). All the Annotations like `@ColorInt`, `@ColorRes` and so on, can all be used without any problems.

I already added some of these annotations myself a while ago but didn't want to run after everyone that doesn't add these with their code.

**Comment by HeinrichReimer** [comment]
@porokoro I know I can use them in my own code without problems but it would be definitely useful if this library declares its parameters using this annotations so that developers aren't confused with colors/color res ids.


For examle this line from [`AppIntroFragment`](https://github.com/PaoloRotolo/AppIntro/blob/master/library/src/main/java/com/github/paolorotolo/appintro/AppIntroFragment.java) should be changed like this:

Old:
````
public static AppIntroFragment newInstance(String title, String description, int imageDrawable, int bgColor) {
````

New:
````
public static AppIntroFragment newInstance(String title, String description, @DrawableRes int imageDrawable, @ColorInt int bgColor) {
````

**Comment by HeinrichReimer** [comment]
Other option would be to use Javadoc to describe accepted parameter values.

**Comment by bubomancer** [comment]
@ja1984 Thanks that worked!

**Comment by Toubap** [comment]
Or you may just be using this : ContextCompat.getColor(this, R.color.primaryColor)

**Comment by HeinrichReimer** [comment]
If you have troubles using color resources as background you may take a look at HeinrichReimer/material-intro. It has some other cool improvements too.

**Comment by avluis** [comment]
Looks like this no longer applies after the recent changes on the library, please reopen this issue if still affected by this issue.
